project/SparkBuild.scala

```diff
@@ -291,6 +291,12 @@ object SparkBuild extends PomBuild {
     publishLocal in SbtCompile := publishTask(publishLocalConfiguration in SbtCompile).value,
     publishLocal := Seq(publishLocal in MavenCompile, publishLocal in SbtCompile).dependOn.value,
 
+    javaOptions ++= {
+      val versionParts = System.getProperty("java.version").split("[+.\\-]+", 3)
+      var major = versionParts(0).toInt
+      if (major >= 16) Seq("--add-modules=jdk.incubator.vector") else Seq.empty
```

Comment:
Would a Spark cluster have to launch its JVMs with this flag to make it work? that's OK if so, we might just add a note in the docs about it.

Reply:
Yes, they would have to. It's one of the failsafe to make sure users don't end up using the feature "by accident". That will be the case until the Vector API becomes stable (gets out of incubation) and doesn't need this flag anymore.